### PR TITLE
mds: no assertion on inode being purging in find_ino_peers()

### PR DESCRIPTION
**Pull request by david-z**
Fixes: #http://tracker.ceph.com/issues/21722

Signed-off-by: Zhi Zhang <zhangz.david@outlook.com>

### CODE REVIEW

File: src/mds/MDCache.cc [hunk 1]
```diff
@@ -8813,7 +8813,8 @@ void MDCache::open_ino(inodeno_t ino, int64_t pool, MDSInternalContextBase* fin,
 void MDCache::find_ino_peers(inodeno_t ino, MDSInternalContextBase *c, mds_rank_t hint)
 {
   dout(5) << "find_ino_peers " << ino << " hint " << hint << dendl;
-  assert(!have_inode(ino));
+  CInode *in = get_inode(ino);
+  assert(!have_inode(ino) || (in && in->state_test(CInode::STATE_PURGING)));
```
Comment (ukernel) [inline src/mds/MDCache.cc]: should be:

if (in && in->state_test(CInode::STATE_PURGING)) {
  c->complete(-ESTALE);
  return;
}
assert(!in)

Comment (david-z) [reply src/mds/MDCache.cc]: good suggestion!

**Comment by david-z** [comment]
@ukernel   Pls review the new update. The "Unmodified Submodules" failure seems not to be related with this change. Thanks.

**Comment by tchaikov** [comment]
retest this please.

**Comment by david-z** [comment]
@ukernel Is this ready to merge? Thanks.

**Comment by david-z** [comment]
@ukernel  ping... :)

**Comment by batrick** [comment]
@david-z I will run it through QA and then merge. (Please don't forget the cephfs label otherwise I won't see this PR.)

**Comment by david-z** [comment]
@batrick got it, thanks.

**Pull request merged**